graph/src/data/query/result.rs

```diff
@@ -45,12 +46,14 @@ pub type Data = Object;
 /// A collection of query results that is serialized as a single result.
 pub struct QueryResults {
     results: Vec<Arc<QueryResult>>,
+    pub validation_time: Option<Duration>,
```

Comment:
Not great that this field had to be added. Could perhaps the GraphQlRunner have access to the `GraphQLServiceMetrics` and register the metric immediately after taking the measurement?

Reply:
I created `GraphQLMetrics` that is passed to `GraphQlRunner` and is responsible for all metrics related to GraphQL:
- `query_execution_time`
- `query_parsing_time`
- `query_validation_time`
- `query_result_size`
- `query_result_max`